### ISSUE TITLE
[Feature Request] L1 L2 Regularization

### ISSUE THREAD

**Issue by well-zt** [deepmodeling/deepmd-kit#4415]
### Summary

Hi, I'm using Deepmd-kit, but I encountered over-fitting problem during training. How can I port L1 L2 regularization into deepmd-kit? 

### Detailed Description

Just as summary

**Comment by caic99** [comment]
Hi @well-zt ,
You can try setting `gradient_max_norm` parameter in input.json. The related codes are here:
https://github.com/deepmodeling/deepmd-kit/blob/d162d0b47b51ea4aee56acbafb0c30fbc1dcdb82/deepmd/pt/train/training.py#L689-L694
[Ref](https://pytorch.org/docs/stable/generated/torch.nn.utils.clip_grad_norm_.html#torch.nn.utils.clip_grad_norm_)

**Comment by QuantumMisaka** [comment]
@caic99 L1, L2 regularization is a common practice in ML training but not common in MLIP training, should we put `how (and why) to use regularization in loss function` in docs of deepmd-kit ?

**Comment by caic99** [comment]
> [@caic99](https://github.com/caic99) L1, L2 regularization is a common practice in ML training but not common in MLIP training, should we put `how (and why) to use regularization in loss function` in docs of deepmd-kit ?

@QuantumMisaka I would like to correct my reply above: what I meant was the L2 norm used for gradient clipping, which is not the same thing as the original poster said.

> how (and why) to use regularization in loss function


Unluckily I have no experience on using regularization for MLIP.
